mypyc/irbuild/expression.py

```diff
@@ -382,6 +382,8 @@ def transform_basic_comparison(builder: IRBuilder,
                                left: Value,
                                right: Value,
                                line: int) -> Value:
+    if is_int_rprimitive(left.type) and is_int_rprimitive(right.type) and op in ('=='):
```

Comment:
The latter part should be `op == '=='` or `op in ('==',)`, as now it's equivalent to `op in '=='`.